### ISSUE TITLE
TextBlocks do not update inline Hyperlinks' IsOffscreen property

### ISSUE THREAD

**Issue by joshkoon** [microsoft/microsoft-ui-xaml#5666]
**Describe the bug**

When TextBlocks containing Hyperlinks are offscreen, the IsOffscreen property of the Hyperlinks is not updated, causing failures in Accessibility Insights:
* "An onscreen element must not have a null BoundingRectangle property"

**Steps to reproduce the bug**

1. Build/Deploy attached HyperlinkTestApp
2. Reduce window size to smallest height
3. Run Accessibility Insights test on the app

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/45607479/129102761-79056263-df4d-41f2-90cf-507c7fdc3b6c.png)

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml 2.6.0]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows version                    | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes   |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

**Attachments**
[HyperlinkTestApp.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6971378/HyperlinkTestApp.zip)

**Comment by StephenLPeters** [comment]
@Austin-Lamb I'm putting this on framework, but I think you know quite a bit about hyperlink, thoughts here?

**Comment by joshkoon** [comment]
@StephenLPeters I noticed you added area-HyperlinkButton, but this only impacts inline Hyperlinks within TextBlocks. HyperlinkButtons behave correctly. Seems to be an issue with the TextBlock not keeping its Inlines up to date, which I believe is only important in the case of Hyperlinks.

**Comment by chrisglein** [comment]
Curious if this is just `Hyperlink` or if other parts under a `TextBlock` that also need to report `IsOffscreen` are reporting correctly.

**Comment by codendone** [comment]
This is an issue in HyperlinkAutomationPeer, which has never implemented the `IsOffscreen` property.

**Comment by RolandLeFranc** [comment]
Hi there, our team also encountered multiple occurrences of this issue when testing our app with Accessibility Insights.
Given that Microsoft requires a pass on Accessibility for OEMs to preload apps in 2022, it would be great to have this bug fixed. Just adding a few cents here.
